datafusion/core/src/datasource/memory.rs

```diff
@@ -648,9 +649,14 @@ mod tests {
         // Create a table scan logical plan to read from the source table
         let scan_plan = LogicalPlanBuilder::scan("source", source, None)?.build()?;
         // Create an insert plan to insert the source data into the initial table
-        let insert_into_table =
-            LogicalPlanBuilder::insert_into(scan_plan, "t", &schema, InsertOp::Append)?
-                .build()?;
+        let insert_into_table = LogicalPlanBuilder::insert_into(
+            scan_plan,
+            "t",
+            table_sink,
```

Comment:
I found the name of `table_sink` to be confusing here (as the type is called `TableSource` I would expect it to be called `table_source` for consistency)

Reply:
But now I see that it is a sink because it is the target of the DML operations

Maybe `target` would be clearer 🤔